drivers/at30tse75x/at30tse75x.c

```diff
@@ -0,0 +1,312 @@
+    i2c_acquire(dev->i2c);
+    if(i2c_read_regs(dev->i2c, dev->addr, reg, (char*) data, 2) <= 0) {
+        DEBUG("[at30tse75x] Can't read register 0x%x\n", reg);
+        i2c_release(dev->i2c);
+        return -1;
+    }
+    i2c_release(dev->i2c);
+
+    return 0;
+}
+
+static int at30tse75x_set_register(at30tse75x_t* dev, uint8_t reg, uint16_t* data)
+{
+    i2c_acquire(dev->i2c);
+    if(i2c_write_regs(dev->i2c, dev->addr, reg, (char*) data, 2) <= 0) {
+        DEBUG("[at30tse75x] Can't write to register 0x%x\n", reg);
+        i2c_release(dev->i2c);
+        return -1;
+    }
+    i2c_release(dev->i2c);
+
+    return 0;
+}
+
+static int at30tse75x_reset(at30tse75x_t* dev)
+{
+    i2c_acquire(dev->i2c);
+    if(i2c_write_byte(dev->i2c, 0x00, AT30TSE75X_CMD__GENERAL_CALL_RESET) != 1) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+    i2c_release(dev->i2c);
+    /* Wait for reset to complete */
+    xtimer_usleep(500);
+    return 0;
+}
+
+int at30tse75x_get_config(at30tse75x_t* dev, uint8_t* data)
+{
+    i2c_acquire(dev->i2c);
+    if(i2c_read_reg(dev->i2c, dev->addr, AT30TSE75X_REG__CONFIG, (char*) data) <= 0) {
+        DEBUG("[at30tse75x] Can't read CONFIG register\n");
+        i2c_release(dev->i2c);
+        return -1;
+    }
+    i2c_release(dev->i2c);
+
+    return 0;
+}
+
+int at30tse75x_set_config(at30tse75x_t* dev, uint8_t data)
+{
+    i2c_acquire(dev->i2c);
+    if(i2c_write_reg(dev->i2c, dev->addr, AT30TSE75X_REG__CONFIG, (char) data) <= 0) {
+        DEBUG("[at30tse75x] Can't write to CONFIG register\n");
+        i2c_release(dev->i2c);
+        return -1;
+    }
+    i2c_release(dev->i2c);
+
+    return 0;
+}
+
+int at30tse75x_set_resolution(at30tse75x_t* dev, at30tse75x_resolution_t resolution)
+{
+    uint8_t config;
+
+    if(resolution < AT30TSE75X_RESOLUTION_9BIT ||
+       resolution > AT30TSE75X_RESOLUTION_12BIT) {
+        return -1;
+    }
+
+    if(at30tse75x_get_config(dev, &config) != 0) {
+        return -1;
+    }
+
+    config &= ~(AT30TSE75X_CONFIG__RESOLUTION_MASK);
+    config |= resolution << AT30TSE75X_CONFIG__RESOLUTION_SHIFT;
+
+    if(at30tse75x_set_config(dev, config) != 0) {
+        return -1;
+    }
+
+    return 0;
+}
+
+int at30tse75x_set_mode(at30tse75x_t* dev, at30tse75x_mode_t mode)
+{
+    uint8_t config;
+    if(at30tse75x_get_config(dev, &config) != 0) {
+        return -1;
+    }
+
+    switch(mode) {
+    case AT30TSE75X_MODE_ONE_SHOT:
+        config |= AT30TSE75X_CONFIG__SHUTDOWN_BIT;
+        /* Don't touch alarm mode */
+        break;
+    case AT30TSE75X_MODE_COMPARATOR:
+        config &= ~(AT30TSE75X_CONFIG__SHUTDOWN_BIT);
+        config &= ~(AT30TSE75X_CONFIG__ALARM_MODE_BIT);
+        break;
+    case AT30TSE75X_MODE_INTERRUPT:
+        config &= ~(AT30TSE75X_CONFIG__SHUTDOWN_BIT);
+        config |= AT30TSE75X_CONFIG__ALARM_MODE_BIT;
+        break;
+    default:
+        return -1;
+    }
+
+    if(at30tse75x_set_config(dev, config) != 0) {
+        return -1;
```

Comment:
Ah okay, didn't get this completely right the first time. Yes I like that, so `-1` is `general i2c failure` and `-2` wrong user input in this case. So `-2` will be returned in the `switch::default` statement. Will update.